### ISSUE TITLE
[Feature Request]: compilation.options.cache missing

### ISSUE THREAD

**Issue by zackarychapple** [web-infra-dev/rspack#2621]
### What problem does this feature solve?

When running the Angular build there is a call to `compilation.options.cache`.

```bash
Failed to initialize Angular compilation - compilation.getCache is not a function
Error: Failed to initialize Angular compilation - compilation.getCache is not a function
    at addError (/Users/packages/ngtools/webpack/src/ivy/diagnostics.ts:35:27)
    at /Users/packages/ngtools/webpack/src/ivy/plugin.ts:192:17
    at Hook.eval (eval at create (/Users/zackarychapple/code/eduardo-rspack/node_modules/tapable/lib/HookCodeFactory.js:19:10), <anonymous>:9:1)
    at Hook.CALL_DELEGATE [as _call] (/Users/zackarychapple/code/eduardo-rspack/node_modules/tapable/lib/Hook.js:14:14)
    at Compiler._Compiler_newCompilation (/Users/zackarychapple/code/eduardo-rspack/packages/rspack/src/compiler.ts:626:30)
```

The call in question is below: 
```ts
if (!this.webpackCache && compilation.options.cache) {
    this.webpackCache = compilation.getCache(PLUGIN_NAME);
}
```

### What does the proposed API of configuration look like?

There are two potential options, either expose getCache, or add documentation on how to set compilation options so caching can be disabled.

**Comment by IWANABETHATGUY** [comment]
Closed in favor of https://github.com/web-infra-dev/rspack/pull/2892

**Comment by zackarychapple** [comment]
Confirmed as resolved by #2892
